### PR TITLE
add patches for mauveAligner to fix build on RHEL8

### PR DESCRIPTION
**Pull request by boegel**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by boegel** [comment]
@boegelbot please test @ generoso

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node3103.skitty.os - Linux CentOS Linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 3.6.8
See https://gist.github.com/d97019712f32cad36b101aade3471e18 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node3524.doduo.os - Linux RHEL 8.4, x86_64, AMD EPYC 7552 48-Core Processor (zen2), Python 3.6.8
See https://gist.github.com/cb8c467479521ac3517428be55154e56 for a full test report.

**Comment by boegel** [comment]
@smoors I'l try to use this PR to test an enhancement to the tests to make the test suite fail if the patch description is missing (that may involve a couple of forced pushes in here).

**Comment by boegel** [comment]
Check for proper patch description works, I fleshed that out into a separate PR: #15061

File: easybuild/easyconfigs/m/mauveAligner/sgEvolver-4736_fix-getopt.patch [hunk 1]
```diff
@@ -0,0 +1,12 @@
+--- sgEvolver/src/getopt.h.orig	2022-02-25 10:33:10.015721000 +0100
```
[the remark this answers is not in the document]
Comment (boegel) [reply easybuild/easyconfigs/m/mauveAligner/sgEvolver-4736_fix-getopt.patch:4]: fixed in 52dacd9

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (1 easyconfigs in total)
node2626.swalot.os - Linux CentOS Linux 7.9.2009, x86_64, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz (haswell), Python 3.6.8
See https://gist.github.com/5f699b6d3bd2c705ad214d0dd673386f for a full test report.

**Comment by smoors** [comment]
Test report by @smoors
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node300.hydra.os - Linux CentOS Linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz (skylake_avx512), Python 2.7.5
See https://gist.github.com/19da039b34f5027a4cde0c8788c9695e for a full test report.

**Review by smoors — APPROVED** [review]
lgtm

**Comment by smoors** [comment]
Going in, thanks @boegel!

**Pull request merged**